### DISCUSSION TITLE
ArgoCD compared to Gitlab CI: Is ArgoCD the next? If yes, please explain.

### DISCUSSION THREAD

**Discussion by jkroepke** [argoproj/argo-cd#5164] [Q&A] [answered]
Hi!

the last days I read a lot about GitOps. I found ArgoCD which helps to sync the (helm-)manifests to an Kubernetes cluster.

Currently we are using classical Gitlab CI which builds docker images then a helm apply or helmfile sync using manifests from the same or other git repository. It looks like GitOps, not?

Maybe I misunderstand something or I just don't get the bug advantage for tools like ArgoCD compared to `helm upgrade` on Gitlab CI. Could someone explain it? 

Thanks.

**Comment by rumstead** [comment]
You are effectively asking CIOps vs GitOps. Weaveworks has a good article on some pros and cons

https://www.weave.works/blog/kubernetes-anti-patterns-let-s-do-gitops-not-ciops

**Comment by jkroepke** [comment, marked as answer]
CIOps was great keyword. I found a article from 2020 which covers the most of my questions.

https://medium.com/flant-com/gitops-2a67a5b82884

Thanks a lot!

**Reply by rumstead** [reply]
No problem!

**Comment by alexmt** [comment]
FYI GitLab is working on [GitLab Kubernetes agent](https://docs.gitlab.com/ee/user/clusters/agent/). Agent implements GitOps flow and based on [Argo GitOps Engine](https://blog.argoproj.io/argo-gitops-engine-update-8a2d6911a25c)

**Reply by jkroepke** [reply]
Just have to wait until it’s avalible in core 😕 nice hint anyway.
